### PR TITLE
Added Junit tests in commons/setting package 

### PR DESCRIPTION
**Pull request by sonja-ct**
Added Junit tests in commons/setting package for:
AbstractBaseKapuaSetting class
SystemSetting class

Signed-off-by: Sonja <sonja.matic@comtrade.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+
+    @Test
+    public void getNullClassEmptyMapTest() {
+        assertNull("Null expected.", abstractBaseKapuaSetting.get(null, key));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getNullClassTest() {
+        map.put("Key", 10);
+        abstractBaseKapuaSetting.get(null, key);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:100]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getNullKeyTest() {
+        abstractBaseKapuaSetting.get(String.class, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:105]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+
+    @Test
+    public void getDefaultValueNullClassEmptyMapTest() {
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.get(null, key, 10));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDefaultValueNullClassTest() {
+        map.put("Key", 10);
+        abstractBaseKapuaSetting.get(null, key, 10);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:124]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 4]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.get(String.class, null, 10);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:129]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 5]
```diff
@@ -0,0 +1,659 @@
...
+    @Test
+    public void getListNullClassEmptyMapTest() {
+        assertThat("Instance of List expected.", abstractBaseKapuaSetting.getList(null, key), IsInstanceOf.instanceOf(List.class));
+        assertTrue("True expected.", abstractBaseKapuaSetting.getList(null, key).isEmpty());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getListNullClassTest() {
+        map.put("Key", 10);
+        abstractBaseKapuaSetting.getList(null, key);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:157]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 6]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getListNullKeyTest() {
+        abstractBaseKapuaSetting.getList(String.class, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:162]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 7]
```diff
@@ -0,0 +1,659 @@
...
+            abstractBaseKapuaSetting.getMap(null, key, "Key");
+            fail("NullPointerException expected.");
+        } catch (Exception e) {
+            assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getMapClassNullKeyRegexTest() {
+        abstractBaseKapuaSetting.getMap(String.class, null, "Key");
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:195]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 8]
```diff
@@ -0,0 +1,659 @@
...
+            abstractBaseKapuaSetting.getMap(null, key);
+            fail("NullPointerException expected.");
+        } catch (Exception e) {
+            assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getMapClassNullKeyTest() {
+        abstractBaseKapuaSetting.getMap(String.class, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:249]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 9]
```diff
@@ -0,0 +1,659 @@
...
+        map.put("Key.key", "Value");
+        Map<String, Object> expectedMap = new HashMap();
+        expectedMap.put("key", "Value");
+
+        assertEquals("Expected and actual values should be the same.", expectedMap, abstractBaseKapuaSetting.getMap(String.class, key));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getIntNullTest() {
+        abstractBaseKapuaSetting.getInt(null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:263]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 10]
```diff
@@ -0,0 +1,659 @@
...
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.getInt(key));
+
+        System.setProperty("Key", "10");
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.getInt(key));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void getIntEmptyMapTest() {
+        abstractBaseKapuaSetting.getInt(key);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:282]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 11]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void getIntDifferentKeyValueTest() {
+        map.put("Other Key", 10);
+
+        abstractBaseKapuaSetting.getInt(key);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:289]: Please add assertion here (i guess assertNull)?

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 12]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getIntNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getInt(key, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:294]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 13]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getIntDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.getInt(null, 10);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:299]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 14]
```diff
@@ -0,0 +1,659 @@
...
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.getInt(key, 10));
+
+        System.setProperty("Key", "10");
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.getInt(key, 10));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getIntNullDefaultValueIntegerClassTest() {
+        abstractBaseKapuaSetting.getInt(key, (Integer) null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:319]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 15]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getIntDefaultValueNullKeyIntegerClassTest() {
+        abstractBaseKapuaSetting.getInt(null, (Integer) 10);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:324]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 16]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.getInt(key, (Integer) 10));
+        System.setProperty("Key", "10");
+        assertEquals("Expected and actual values should be the same.", 10, abstractBaseKapuaSetting.getInt(key, (Integer) 10));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getBooleanNullTest() {
+        abstractBaseKapuaSetting.getBoolean(null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:343]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 17]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", true, abstractBaseKapuaSetting.getBoolean(key));
+        System.setProperty("Key", "true");
+        assertEquals("Expected and actual values should be the same.", true, abstractBaseKapuaSetting.getBoolean(key));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getBooleanNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getBoolean(key, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:361]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 18]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getBooleanDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.getBoolean(null, true);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:366]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 19]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", true, abstractBaseKapuaSetting.getBoolean(key, true));
+        System.setProperty("Key", "true");
+        assertEquals("Expected and actual values should be the same.", true, abstractBaseKapuaSetting.getBoolean(key, true));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getBooleanNullDefaultValueBooleanClassTest() {
+        abstractBaseKapuaSetting.getBoolean(key, (Boolean) null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:385]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 20]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getBooleanDefaultValueNullKeyBooleanClassTest() {
+        abstractBaseKapuaSetting.getBoolean(null, (Boolean) true);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:390]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 21]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", true, abstractBaseKapuaSetting.getBoolean(key, (Boolean) true));
+        System.setProperty("Key", "true");
+        assertEquals("Expected and actual values should be the same.", true, abstractBaseKapuaSetting.getBoolean(key, (Boolean) true));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getStringNullTest() {
+        abstractBaseKapuaSetting.getString(null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:408]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 22]
```diff
@@ -0,0 +1,659 @@
...
+    @Test
+    public void getStringNullDefaultValueTest() {
+        assertNull("Null expected.", abstractBaseKapuaSetting.getString(key, null));
+        map.put("Key", "Value");
+        assertEquals("Expected and actual values should be the same.", "Value", abstractBaseKapuaSetting.getString(key, null));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getStringDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.getString(null, "Value");
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:433]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 23]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", "Value", abstractBaseKapuaSetting.getString(key, "Value"));
+        System.setProperty("Key", "Value");
+        assertEquals("Expected and actual values should be the same.", "Value", abstractBaseKapuaSetting.getString(key, "Value"));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongNullTest() {
+        abstractBaseKapuaSetting.getLong(null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:452]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 24]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getLong(key, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.getLong(null, 10L);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:475]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 25]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", 10L, abstractBaseKapuaSetting.getLong(key));
+        System.setProperty("Key", "10");
+        assertEquals("Expected and actual values should be the same.", 10L, abstractBaseKapuaSetting.getLong(key));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getLong(key, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:470]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 26]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongDefaultValueNullKeyLongClassTest() {
+        abstractBaseKapuaSetting.getLong(null, (Long) 10L);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatNullTest() {
+        abstractBaseKapuaSetting.getFloat(null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:516]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 27]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongNullDefaultValueLongClassTest() {
+        abstractBaseKapuaSetting.getLong(key, (Long) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongDefaultValueNullKeyLongClassTest() {
+        abstractBaseKapuaSetting.getLong(null, (Long) 10L);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:511]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 28]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", 10L, abstractBaseKapuaSetting.getLong(key, (Long) 10L));
+        System.setProperty("Key", "10");
+        assertEquals("Expected and actual values should be the same.", 10L, abstractBaseKapuaSetting.getLong(key, (Long) 10L));
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getLongNullDefaultValueLongClassTest() {
+        abstractBaseKapuaSetting.getLong(key, (Long) null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:506]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 29]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getFloat(key, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.getFloat(null, 10.11f);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:540]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 30]
```diff
@@ -0,0 +1,659 @@
...
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getFloat(key), 0.2);
+        System.setProperty("Key", "10.11");
+
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getFloat(key), 0.2);
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getFloat(key, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:535]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 31]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatNullDefaultValueFloatClassTest() {
+        abstractBaseKapuaSetting.getFloat(key, (Float) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatDefaultValueNullKeyFloatClassTest() {
+        abstractBaseKapuaSetting.getFloat(null, (Float) 10.11f);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:563]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 32]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getFloat(key, 10.11f), 0.2);
+        System.setProperty("Key", "10.11");
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getFloat(key, 10.11f), 0.2);
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFloatNullDefaultValueFloatClassTest() {
+        abstractBaseKapuaSetting.getFloat(key, (Float) null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:558]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 33]
```diff
@@ -0,0 +1,659 @@
...
+        System.setProperty("Key", "10.11");
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getFloat(key, (Float) 10.11f), 0.2);
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleNullTest() {
+        map.put("Key", 10.11);
+
+        abstractBaseKapuaSetting.getDouble(null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:583]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 34]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getDouble(key, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleDefaultValueNullKeyTest() {
+        abstractBaseKapuaSetting.getDouble(null, 10.11);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:607]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 35]
```diff
@@ -0,0 +1,659 @@
...
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getDouble(key), 0.2);
+        System.setProperty("Key", "10.11");
+
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getDouble(key), 0.2);
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleNullDefaultValueTest() {
+        abstractBaseKapuaSetting.getDouble(key, null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:602]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 36]
```diff
@@ -0,0 +1,659 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleNullDefaultValueDoubleClassTest() {
+        abstractBaseKapuaSetting.getDouble(key, (Double) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleDefaultValueNullKeyDoubleClassTest() {
+        abstractBaseKapuaSetting.getDouble(null, (Double) 10.11);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:630]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 37]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getDouble(key, 10.11), 0.2);
+        System.setProperty("Key", "10.11");
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getDouble(key, 10.11), 0.2);
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getDoubleNullDefaultValueDoubleClassTest() {
+        abstractBaseKapuaSetting.getDouble(key, (Double) null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:625]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 38]
```diff
@@ -0,0 +1,659 @@
...
+
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getDouble(key, (Double) 10.11), 0.2);
+        System.setProperty("Key", "10.11");
+        assertEquals("Expected and actual values should be the same.", 10.11, abstractBaseKapuaSetting.getDouble(key, (Double) 10.11), 0.2);
+        System.clearProperty("Key");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void isSystemPropertyHotSwapNullTest() {
+        abstractBaseKapuaSetting.setSystemPropertyHotSwap((Boolean) null);
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:648]: Please add assertion here.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/system/SystemSettingTest.java [hunk 39]
```diff
@@ -0,0 +1,41 @@
...
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.commons.setting.system;
+
+import org.eclipse.kapua.qa.markers.junit.JUnitTests;
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/system/SystemSettingTest.java:19]: Please delete empty line.

File: commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java [hunk 40]
```diff
@@ -0,0 +1,659 @@
...
+
+import org.apache.commons.configuration.DataConfiguration;
+import org.apache.commons.configuration.MapConfiguration;
+import org.eclipse.kapua.qa.markers.junit.JUnitTests;
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
```
Comment (LeoNerdoG) [inline commons/src/test/java/org/eclipse/kapua/commons/setting/AbstractBaseKapuaSettingTest.java:22]: Please delete empty line.

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
hej @sonja-ct,
I have reviewed this PR and it looks OK, there are only few minor things that need to be changed/added (mostly assertions in some methods) and some lines to be deleted. 
Please take a look 😄

**Review by LeoNerdoG — APPROVED** [review]
Hey @Coduz,
I have checked this PR and it looks OK, can you please merge it when CI passes? 
Thanks!

**Review by Coduz — APPROVED** [review]

**Pull request merged**